python/apps/taiga/src/taiga/workflows/services/__init__.py

```diff
@@ -128,7 +129,60 @@ async def update_workflow_status(
     updated_status = await workflows_repositories.update_workflow_status(workflow_status=workflow_status, values=values)
 
     await workflows_events.emit_event_when_workflow_status_is_updated(
-        project=workflow.project, workflow_status=updated_status
+        project=workflow_status.project, workflow_status=updated_status
     )
 
     return updated_status
+
+
+##########################################################
+# delete workflow status
+##########################################################
+
+
+async def delete_workflow_status(workflow_status: WorkflowStatus, move_to_status_slug: str | None) -> bool:
+    """
+    This method deletes a workflow status, providing the option to first migrating its stories to another workflow
+    status of the same workflow.
+
+    :param workflow_status: the workflow status to delete
+    :param move_to_status_slug: the workflow status's slug to which move the stories from the status being deleted
+        - if not received, all the workflow status and its contained stories will be deleted
+        - if received, the workflow status will be deleted but its contained stories won't (they will be first moved to
+         the specified status)
+    :return: None
+    """
+    if move_to_status_slug:
+        move_to_status = await get_workflow_status(
+            project_id=workflow_status.project.id,
+            workflow_slug=workflow_status.workflow.slug,
+            status_slug=move_to_status_slug,
+        )
+        if not move_to_status:
+            raise ex.NonExistingMoveToStatus(f"The status '{move_to_status_slug}' doesn't exist")
+        if move_to_status == workflow_status:
+            raise ex.SameMoveToStatus("The stories can't be moved to the workflow status being deleted")
```

Comment:
I think it's better: "The to-be-deleted status and the target-status cannot be the same"

Reply:
Thak you I were struggling a lot chosing a name for this exception. I buy it for sure!  :rose: